Control/Monad/Writer/Class.hs

```diff
@@ -205,3 +212,80 @@ instance
     tell   = lift . tell
     listen = Accum.liftListen listen
     pass   = Accum.liftPass pass
+
+
+-- | A helper type to decrease boilerplate when defining new transformer
+-- instances of 'MonadWriter'.
+--
+-- @since ????
+type LiftingWriter :: ((Type -> Type) -> Type -> Type) -> (Type -> Type) -> Type -> Type
+newtype LiftingWriter t m a = LiftingWriter {runLiftingWriter :: t m a}
+  deriving (Functor, Applicative, Monad, MonadTrans)
+
+
+instance (Monoid w', MonadWriter w m) => MonadWriter w (LiftingWriter (LazyRWS.RWST r w' s) m) where
+  writer = lift . writer
+  tell = lift . tell
+  listen (LiftingWriter (LazyRWS.RWST x)) = LiftingWriter $ LazyRWS.RWST $ \r s -> do
+    ((a, s, w'), w) <- listen $ x r s
+    pure ((a, w), s, w')
+  pass (LiftingWriter (LazyRWS.RWST x)) = LiftingWriter $ LazyRWS.RWST $ \r s -> do
+    (y, s, w') <- x r s
+    a <- pass $ pure y
+    pure (a, s, w')
```

Comment:
These `pass` implementations are incorrect. Here is a test:

```hs
newtype SneakyWriterT m a = SneakyWriterT { runSneakyWriterT :: Lazy.WriterT String m a }
  deriving (Functor, Applicative, Monad)
  deriving (MonadWriter w) via LiftingWriter (Lazy.WriterT String) m

main :: IO ()
main = print $ runIdentity $ Lazy.runWriterT $ Lazy.runWriterT $ runSneakyWriterT $ pass $ do
  tell "hello"
  pure ((), reverse)
```

This should print `(((),""),"olleh")`, but with your implementation it prints `(((),""),"hello")`.

Avoiding errors like this is one reason why I [suggested](https://github.com/haskell/mtl/pull/149#pullrequestreview-2009506739) using the existing `map*T` functions instead of inlining your own implementations of them. For `MonadWriter`, that would look like:

```hs
sequenceFst :: ((a, b), c) -> ((a, c), b)
sequenceFst ((a, b), c) = ((a, c), b)

sequenceFst23 :: ((a, b, c), d) -> ((a, d), b, c)
sequenceFst23 ((a, b, c), d) = ((a, d), b, c)

sequenceFst32 :: ((a, b), c, d) -> ((a, c, d), b)
sequenceFst32 ((a, b), c, d) = ((a, c, d), b)

mapLiftingWriter :: (t m a -> t m b) -> LiftingWriter t m a -> LiftingWriter t m b
mapLiftingWriter = coerce

instance (Monoid w', MonadWriter w m) => MonadWriter w (LiftingWriter (LazyRWS.RWST r w' s) m) where
  writer = lift . writer
  tell = lift . tell
  listen = mapLiftingWriter (LazyRWS.mapRWST (fmap sequenceFst23 . listen))
  pass = mapLiftingWriter (LazyRWS.mapRWST (pass . fmap sequenceFst32))

-- ... and likewise for the remaining RWST instances

instance (Monoid w', MonadWriter w m) => MonadWriter w (LiftingWriter (Lazy.WriterT w') m) where
  writer = lift . writer
  tell = lift . tell
  listen = mapLiftingWriter (Lazy.mapWriterT (fmap sequenceFst . listen))
  pass = mapLiftingWriter (Lazy.mapWriterT (pass . fmap sequenceFst))

-- ... and likewise for the remaining WriterT instances
```